### ISSUE TITLE
Why not use "softmax" function to replace the "sigmoid" function

### ISSUE THREAD

**Issue by yianzhongguo** [ellisdg/3DUnetCNN#148]
I found the sigmoid function was used in the output of isensee2017 model. When I use the softmax function to replace it, the training prcoess ended with the resut: the loss was about -0.04. It puzzled me.

**Comment by CocoInParis** [comment]
same question here, and personally thought softmax might be more suitable for this problem. did you figure this out?
or we could discuss this.
thank you!

**Comment by parvezamu** [comment]
If every example can be associated with multiple labels, you need to use a sigmoid output layer that learns to predict "yes" or "no" for each individual label. If the classes are disjoint, i.e. each example can only belong to one class, you should use a softmax output layer to incorporate this constraint. source: https://www.reddit.com/r/MachineLearning/comments/32iyt9/question_comparison_between_softmax_and_sigmoid/
Another source: https://dev.to/rosejcday/learning-the-differences-between-softmax-and-sigmoid-for-image-classification--59c

**Comment by Paddy-Xu** [comment]
sorry to reopen this again 3 years later. But I am also confused about it and has not find the reason why? Could you please have a look? Thanks a lot!

**Comment by ellisdg** [comment]
If I remember correctly, for softmax you have to encode background as an additional label. I haven't done it in awhile, but you should be able to add 0 to the list of labels and then change the activation function to softmax. You can do all this in the config json file that you use for training.
